applications/ContactStructuralMechanicsApplication/python_scripts/auxiliar_methods_solvers.py

```diff
@@ -235,14 +231,20 @@ def AuxiliarCreateLinearSolver(main_model_part, settings, contact_settings, line
                 mixed_ulm_solver = CSMA.MixedULMLinearSolver(linear_solver, contact_settings["mixed_ulm_solver_parameters"])
                 return mixed_ulm_solver
             else:
-                print_on_rank_zero("::[Contact Mechanical Solver]:: ", "Mixed solver not available: " + name_mixed_solver + ". Using not mixed linear solver")
+                KM.Logger.PrintInfo("::[Contact Mechanical Solver]:: ", "Mixed solver not available: " + name_mixed_solver + ". Using not mixed linear solver")
                 return linear_solver
         else:
             return linear_solver
     else:
         return linear_solver
 
-def AuxiliarLineSearch(computing_model_part, mechanical_scheme, linear_solver, mechanical_convergence_criterion, builder_and_solver, settings, contact_settings, processes_list, post_process):
+def  AuxiliarPureSlipCheck(model_part):
+    # Check if the properties have friction coefficient
+    aux_sum_friction_coeff = sum(prop.GetValue(KM.FRICTION_COEFFICIENT) for prop in model_part.GetProperties() if prop.Has(KM.FRICTION_COEFFICIENT))
+    # Check value
+    return False if aux_sum_friction_coeff > sys.float_info.epsilon else True
```

Comment:
```suggestion
    return aux_sum_friction_coeff < sys.float_info.epsilon
```

would be even shorter but up to you